frontend/src/components/VestingTable.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { CONTRACT_ADDR } from '../lib/config';
+import axios from 'axios';
+import { useAccount } from '@starknet-react/core';
+import { useQuery } from '@tanstack/react-query';
+import { BASE_API_URL } from "../lib/config";
+
+interface VestingEvent {
+    amount: number;
+    claimable_at: number | null;
+    is_claimable: boolean;
+    is_claimed: boolean;
+}
+
+const ITEMS_PER_PAGE = 10; // Items per page
+
+const VestingTable: React.FC = () => {
+    const { address } = useAccount();
+    const [currentPage, setCurrentPage] = useState(1); // Track the current page
+
+    const { data: events = [], isLoading, error } = useQuery({
+        queryKey: ['vesting-events', address],
+        queryFn: async () => {
+            if (!address) {
+                return [];
+            }
+            const response = await axios.get(`${BASE_API_URL}vesting-events`, {
```

Comment:
see `apiService.ts` line 10:

`${BASE_API_URL}/submit`

Unify it in both places such that one BASE_API_URL can work for both.